server/leaderboard/views.py

```diff
@@ -0,0 +1,65 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from .models import LeetcodeStats
+from .serializers import LeetcodeStatsSerializer
+from django.db.models import F, ExpressionWrapper, FloatField
+from rest_framework.exceptions import ValidationError
+from django.utils import timezone
+from datetime import timedelta
+from django.core.management import call_command
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LeetcodeLeaderboardView(generics.ListAPIView):
+    serializer_class = LeetcodeStatsSerializer
+    permission_classes = [IsAuthenticatedOrReadOnly]
+
+    def get_queryset(self):
+        latest_stat = LeetcodeStats.objects.order_by("-last_updated").first()
+        if not latest_stat or timezone.now() - latest_stat.last_updated > timedelta(
+            hours=1
+        ):
+            logger.info("Updating Leetcode stats")
+            call_command("update_leetcode_stats", timeout=2)
```

Comment:
Does it wait for this? will it be very slow if we have a lot of users?

Reply:
If it doesn't wait for it, will we have very outdated info then?

Reply:
yep, could set up cron but I was lazy and didn't want to make deployment more complicated

Reply:
I will add TODO though